src/controllers/tests/validate.controller.test.ts

```diff
@@ -0,0 +1,213 @@
+import request from 'supertest';
+
+import { App } from '../../app';
+import { ProblemException } from '../../exceptions/problem.exception';
+
+import { ValidateController } from '../validate.controller';
+
+const validJSONAsyncAPI = {
+  asyncapi: '2.0.0',
+  info: {
+    title: 'My API',
+    version: '1.0.0',
+  },
+  channels: {
+    '/test/tester': {
+      subscribe: {
+        message: {
+          payload: {
+            type: 'object',
+            properties: {
+              name: {
+                type: 'string',
+                'x-parser-schema-id': '<anonymous-schema-2>',
+              },
+            },
+            'x-parser-schema-id': '<anonymous-schema-1>',
+          },
+          'x-parser-original-schema-format':
+            'application/vnd.aai.asyncapi;version=2.0.0',
+          'x-parser-original-payload': {
+            type: 'object',
+            properties: {
+              name: {
+                type: 'string',
+              },
+            },
+          },
+          schemaFormat: 'application/vnd.aai.asyncapi;version=2.0.0',
+          'x-parser-message-parsed': true,
+          'x-parser-message-name': '<anonymous-message-1>',
+        },
+      },
+    },
+  },
+  'x-parser-spec-parsed': true,
+};
```

Comment:
It's not a valid AsyncAPI spec 😅  I mean it is, but it's also parsed and validated spec (see extensions `x-**` fields), so sending that is not a good idea, you should send this:

```json
{
  "asyncapi": "2.2.0",
  "info": {
    "title": "Account Service",
    "version": "1.0.0",
    "description": "This service is in charge of processing user signups"
  },
  "channels": {
    "user/signedup": {
      "subscribe": {
        "message": {
          "$ref": "#/components/messages/UserSignedUp"
        }
      }
    }
  },
  "components": {
    "messages": {
      "UserSignedUp": {
        "payload": {
          "type": "object",
          "properties": {
            "displayName": {
              "type": "string",
              "description": "Name of the user"
            },
            "email": {
              "type": "string",
              "format": "email",
              "description": "Email of the user"
            }
          }
        }
      }
    }
  }
}
```

It's a valid JSON AsyncAPI but before validation :)

Reply:
yes much better, thank you 😄